### PR TITLE
tools: Make "check-changelogger-use" a proper tool

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

Move .github/files/require-change-file-for-touched-projects.php to
tools/check-changelogger-use.php, pass base and head refs via the
command line instead of reading the GitHub Actions environment, and use
a more human-friendly output when not running under CI.

#### Jetpack product discussion

Code review on #20303

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Is CI happy, and showing proper output for the relevant job?
* Make a commit that touches a project, without adding a change file. Run the script, e.g. `tools/check-changelogger-use.php origin/master HEAD`. Does the output look sensible?
* Run the script again with CI set, e.g. `CI=1 tools/check-changelogger-use.php origin/master HEAD`. Does the output look good for Actions?
* Add a change file, then run it again. Does it no longer complain?

### CODE REVIEW

File: tools/check-changelogger-use.php [hunk 1]
```diff
@@ -128,15 +173,31 @@ function debug() {
...
+				"::error::Project %s is being changed, but no change file in %s is touched!%%0A%%0AUse `jetpack changelogger add %s` to add a change file.%%0AGuidelines: https://github.com/Automattic/jetpack/blob/master/docs/writing-a-good-changelog-entry.md\n",
+				$slug,
+				"projects/$slug/{$changelogger_projects[ $slug ]['changes-dir']}/",
+				$slug
+			);
+			printf( "---\n" );
+			$exit = 1;
+		} else {
+			printf(
+				"\e[1;31mProject %s is being changed, but no change file in %s is touched!\e[0m\n",
```
Comment (sdixon194) [inline tools/check-changelogger-use.php:190]: Would saying `was added` make more sense?

**Comment by sdixon194** [comment]
This works! Just a slight nitpick about wording in case we want to adjust that.

File: tools/check-changelogger-use.php (hunk 1 above)
Comment (anomiex) [reply tools/check-changelogger-use.php:190]: The code intentionally allows modifications and deletions to satisfy the check.

Deletions in particular are counted so as to not require special logic to try to detect a release PR like #20214. If deletions didn't count (and we didn't figure out some other way to detect a release PR), the modifications to `projects/plugins/jetpack/composer.json`, `projects/plugins/jetpack/jetpack.php`, `projects/plugins/jetpack/package.json`, and `projects/plugins/jetpack/readme.txt` in that PR would have required a change entry, which in turn would have required unwanted "-alpha" version bumps in several of those files.

Comment (sdixon194) [reply tools/check-changelogger-use.php:190]: Ah gotcha, thanks for clarifying.

**Review by sdixon194 — APPROVED** [review]

**Pull request merged**